### PR TITLE
Add option to use ASDF in makefile

### PR DESCRIPTION
**Pull request by jacquelineIO**
## Description

MilMove devs are already using ASDF. This adds the the option to continue using ASDF instead of pyenv.

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?

## Setup

Command to get the Customer app running for load testing:

Startup milmove server:
```sh
make server_run
```

Follow the README to setup you env. To use ASDF you can do the following:

Note: python 3.8.3 failed to install on my machine so I actually used latest which was 3.9.5. To bypass using 3.8.3 in the Makefile you can manually install python instead of letting the Makefile do it. Try out the Makefile first and if it fails then move on to do it manually and then run the Makefile again. 

```sh
brew install openssl readline sqlite3 xz zlib
asdf plugin add python
USE_ASDF=true make  venv
# follow directions and activate your env
USE_ASDF=true make  setup
```

### Manually installing python with ASDF

```sh
asdf install python latest
asdf local python 3.9.5 
USE_ASDF=true PY_VERSION=3.9.5 make  venv
# follow directions and activate your env
USE_ASDF=true make  setup
```

Running the master repo and NOT this branch. You can run the following just to see the app open up.
```sh
make load_test_milmove
```

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-8058) for this change.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,7 +6,21 @@ endif
...
+ifndef USE_ASDF
+		USE_ASDF:=false
+endif
+
+# ASDF versions
+# global $HOME/.tool-versions
+# local $PWD/.tool-versions
+# $(HOME)/.asdf/installs/python/$(PY_VERSION)
+# PYENV versions
+# $(HOME)/.pyenv/versions/$(PY_VERSION)
```
Comment (sandy-wright) [inline Makefile:18]: What does this commented code mean?

**Review by sandy-wright — COMMENTED** [review]
This looks good, and I had no issues with it! Would you be willing to add the instructions for this setup to the README? Instead of the `pyenv` section, we could call it "Python versioning" and include both options

**Comment by sandy-wright** [comment]
> This looks good, and I had no issues with it! Would you be willing to add the instructions for this setup to the README? Instead of the `pyenv` section, we could call it "Python versioning" and include both options

Oh, also! Since this project also makes use of `direnv`, we can add the option to put `export USE_ASDF=true` in their `.envrc.local` file so that they don't have to remember to prefix the commands with that.

**Comment by jacquelineIO** [comment]
> This looks good, and I had no issues with it! Would you be willing to add the instructions for this setup to the README? Instead of the `pyenv` section, we could call it "Python versioning" and include both options

@sandy-wright yes, I should have updated the README! I will do that.

File: Makefile (hunk 1 above)
Comment (jacquelineIO) [reply Makefile:18]: @sandy-wright  I was working out where things were stored, probably not needed here.

File: README.md [hunk 2]
```diff
@@ -220,6 +226,56 @@ make teardown
 
 Remember to recreate your virtual environment with `make venv` before attempting to continue development on the project.
 
+### Alternative Setup: ASDF
+
+It is possible to run with `ASDF` instead of using `pyenv`. The `Makefile` file has been updated to check for the env variable
+`USE_ASDR=true`.
```
Comment (sandy-wright) [inline README.md:232]: Small typo - should be `USE_ASDF`

File: README.md [hunk 3]
```diff
@@ -220,6 +226,56 @@ make teardown
...
+
+Add `USE_ASDF` to your `.envrc.local` file.
+
+```shell script
+export USE_ASDR=true
```
Comment (sandy-wright) [inline README.md:237]: Same typo here. I keep thinking of ADRs now lol

**Review by sandy-wright — APPROVED** [review]
LGTM, thanks for updating the README! Commented on a couple of typos, but I'm gonna go ahead and approve so you can merge whenever you get to fix those 😄

**Comment by jacquelineIO** [comment]
> LGTM, thanks for updating the README! Commented on a couple of typos, but I'm gonna go ahead and approve so you can merge whenever you get to fix those 😄

geesh! thanks. I had made this typo several times.

**Pull request merged**